docs/ldml/tr35-personNames.md

```diff
@@ -41,26 +41,924 @@ The LDML specification is divided into the following parts:
+| monogram   | Requests initial grapheme. Example: A name such as<br>{ given: “Landon”, given2: “Bainard Crawford”, surname: “Johnson” }<br>could be represented as<br>“{given-monogram-allCaps}{given2-monogram-allCaps}{surname-monogram-allCaps}”<br>or “**LBJ**”<br><br>_The default implementation uses the first grapheme cluster of the value for the field; if the PersonName object has a locale, and CLDR supports a locale-specific grapheme cluster algorithm for that locale, then that algorithm is used. The PersonName object can override this, as detailed below. The difference between monogram an initial is that monogram only returns one element, not one element per word._<br><br>Only the “_-initial_” or the “_-monogram_” modifier may be used, but not both. They are mutually exclusive. |
+
+There may be more modifiers in the future. 
+
+Examples: 
+
+1. For the initial of the surname “**_de Souza_**”, in a language that treats the “de” as a tussenvoegsel, the PersonName object can automatically recast `{surname-initial}` to:<br>`{surname-prefix-initial}{surname-core-initial-allCaps} `to get “dS” instead of “d”.
+2. If the locale expects a surname prefix to to be sorted after a surname, then both `{surname-core} `then `{surname-prefix}` would be used as in<br>`{surname-core}, {given} {given2} {surname-prefix}`
+
+## 6 <a name="6-formatting-process" href="#6-formatting-process">Formatting Process</a>
+
+The patterns are in personName elements, which are themselves in a personNames container element. The following describes how these patterns are chosen. If the name locale is different than the formatting locale, then additional processing needs to take place: see [Handling foreign names](#6-8-handling-foreign-names).
+
+The details of the XML structure behind the data referenced here are in [XML Structure](#2-xml-structure).
+
+### 6.1 <a name="6-1-derive-the-name-locale" href="#6-1-derive-the-name-locale">Derive the name locale</a>
+
+If there is no name locale available via the PersonName data interface, then an effective name locale is created based on the characters in the name and the formatting locale. 
+
+* Get the predominant script of the given and surname, and find the likely language as specified by the likely subtags. Use the locale identifier constructed from the likely language and the script as the name locale.
```

Comment:
I think we left out a step here.  If the name doesn't specify a locale, we check the name's script, and if the name's script is supported by the formatting locale, the name locale becomes the formatting locale.  If the name's script is _different_ from the formatting locale, _then_ we use the likely-subtags data to derive a name locale.

I'm also wondering if we want to nail down what we mean by the "predominant script" or just leave that unspecified for now.

Reply:
Good catches.

I suggest: 

If there is no name locale available via the PersonName data interface, then an effective name locale is created based on the characters in the name and the formatting locale. 

* Find the predominant script for the name in the following way.
  * For each character in the given and surname, find the script(s) of the character using the Script_Extensions property.
    * For each of those scripts, increment a counter for that script, and record the position of the first character encountered in that script.
  * The predominant script is the script with the highest counter value. If there are multiple counters, take the one with the lowest position. If there is still more than one, take the script whose script code is alphabetically lowest (this step is only needed in edge cases).

* If the predominant script is the same as the script of the formatting locale, then the name locale becomes the formatting locale.

* Otherwise, find the likely locale for the predominant script as specified by the likely subtags. (This will add a language and region.) Use that locale as the name locale.

Reply:
The "alphabetically lowest" thing feels a little weird to me, but hopefully it'll never happen in practice.  Other than that, this looks great.

Reply:
Ya, that is just to give a determinant result. We could have a footnote to that effect.